core/perf_test/PerfTest_ViewCopy.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 3.0
+//       Copyright (2020) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include <PerfTest_ViewCopy.hpp>
+
+namespace Test {
+
+/**
+ * \brief Mark the label as a figure of merit.
+ */
+std::string benchmark_fom(const std::string& label) { return "FOM: " + label; }
+
+void report_results(benchmark::State& state, std::size_t num_elems,
+                    double time) {
+  state.SetIterationTime(time);
+
+  // data size in megabytes
+  const auto size = 1.0 * num_elems * sizeof(double) / 1024 / 1024;
+  // throughput in gigabytes per second
+  const auto throughput = 2 * size / 1024 / time;
+
+  state.counters["MB"] = benchmark::Counter(size, benchmark::Counter::kDefaults,
+                                            benchmark::Counter::OneK::kIs1024);
+  state.counters[benchmark_fom("GB/s")] =
+      benchmark::Counter(throughput, benchmark::Counter::kDefaults,
+                         benchmark::Counter::OneK::kIs1024);
```

Comment:
Can you comment (by answering on this thread) on our computing the throughput ourselves rather than relying on the google benchmark native custom counter as rate support?  Is it about forcing the reporting in GB/s?

Reply:
Good catch, it's possible to use benchmark's machinery. Sample output after 9df3555:

```
----------------------------------------------------------------------------------------------------------------------------------------------
Benchmark                                                                              Time             CPU   Iterations  FOM: GB/s         MB
----------------------------------------------------------------------------------------------------------------------------------------------
ViewDeepCopy_Rank1<Kokkos::LayoutLeft, Kokkos::LayoutLeft>/N:10/manual_time        0.097 s         0.089 s             8  15.3535/s    762.939
```
Column header stays the same (`FOM: GB/s`), the value is reported as "per second" (`15.3535/s`).